src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx

```diff
@@ -360,22 +362,27 @@ export class ProsemirrorAdapter {
         this.view.dom.dispatchEvent(actionBarEvent);
     };
 
-    private handleCancelLinkMenu = () => {
+    private handleCancelLinkMenu = (event: CustomEvent<void>) => {
+        event.preventDefault();
+        event.stopPropagation();
+
         this.isLinkMenuOpen = false;
     };
 
     private handleSaveLinkMenu = () => {
-        this.isLinkMenuOpen = false;
-
-        const saveLinkEvent = new CustomEvent('saveLinkMenu', {
-            detail: {
-                type: EditorMenuTypes.Link,
-                link: this.link,
-            },
+        requestAnimationFrame(() => {
+            this.isLinkMenuOpen = false;
+
+            const saveLinkEvent = new CustomEvent('saveLinkMenu', {
+                detail: {
+                    type: EditorMenuTypes.Link,
+                    link: this.link,
+                },
+            });
+            this.view.dom.dispatchEvent(saveLinkEvent);
+
+            this.link = { href: '' };
```

Comment:
RequestAnimationFram again. Maybe it's a very good reason for it but please explain.

Reply:
This is left over from when we had some quirks with the menu-surface and I wanted to be certain that the DOM was fully updated before handling certain events but we've removed the menu-surface. I had meant to remove this. The only place I feel we should keep it is where we set focus to the first input in the link-menu.